main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -509,6 +510,63 @@ object CompletionProvider {
     }
   }
 
+  private def caseCompletion(label: String, completion: String)(implicit context: Context): CompletionItem = {
+    CompletionItem(label = label,
+        sortText = Priority.high(label),
+        textEdit = TextEdit(context.range, completion),
+        documentation = None,
+        insertTextFormat = InsertTextFormat.Snippet,
+        kind = CompletionItemKind.EnumMember)
+  }
+
+  private def enumCompletionAcc(acc: List[CompletionItem], enm: TypedAst.Enum, currentWordIsCase: Boolean)(implicit context: Context, flix: Flix): List[CompletionItem] = {
+    enm.cases.foldLeft(acc)({
+      case (acc, (sym , cas)) => {
+        val name = sym.name
+        val tpe = cas.tpe
+        val typeString = FormatType.formatType(tpe) match {
```

Comment:
This relies on FormatType, but that could change. You should also use typeConstructor here.